x/subspaces/keeper/grpc_query.go

```diff
@@ -0,0 +1,154 @@
+		if err != nil {
+			return err
+		}
+
+		groups = append(groups, group)
+		return nil
+	})
+
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	return &types.QueryUserGroupsResponse{Groups: groups, Pagination: pageRes}, nil
+}
+
+// UserGroup implements the Query/UserGroup gRPC method
+func (k Keeper) UserGroup(ctx context.Context, request *types.QueryUserGroupRequest) (*types.QueryUserGroupResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	// Check if the subspace exists
+	if !k.HasSubspace(sdkCtx, request.SubspaceId) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "subspace with id %d not found", request.SubspaceId)
+	}
+
+	// Get the group
+	group, found := k.GetUserGroup(sdkCtx, request.SubspaceId, request.GroupId)
+	if !found {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "group %d could not be found", request.GroupId)
+	}
+
+	return &types.QueryUserGroupResponse{Group: group}, nil
+}
+
+// UserGroupMembers implements the Query/UserGroupMembers gRPC method
+func (k Keeper) UserGroupMembers(ctx context.Context, request *types.QueryUserGroupMembersRequest) (*types.QueryUserGroupMembersResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	// Check if the subspace exists
+	if !k.HasSubspace(sdkCtx, request.SubspaceId) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "subspace with id %d not found", request.SubspaceId)
+	}
+
+	// Check if the group exists
+	if !k.HasUserGroup(sdkCtx, request.SubspaceId, request.GroupId) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "group %d could not be found", request.GroupId)
+	}
+
+	store := sdkCtx.KVStore(k.storeKey)
+	storePrefix := types.GroupMembersStoreKey(request.SubspaceId, request.GroupId)
+	membersStore := prefix.NewStore(store, storePrefix)
+
+	var members []string
+	pageRes, err := query.Paginate(membersStore, request.Pagination, func(key []byte, value []byte) error {
+		member := types.GetAddressFromBytes(bytes.TrimPrefix(key, storePrefix))
+		members = append(members, member.String())
+		return nil
+	})
+
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	return &types.QueryUserGroupMembersResponse{Members: members, Pagination: pageRes}, nil
+}
+
+// UserPermissions implements the Query/UserPermissions gRPC method
+func (k Keeper) UserPermissions(ctx context.Context, request *types.QueryUserPermissionsRequest) (*types.QueryUserPermissionsResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	// Check if the subspace exists
+	if !k.HasSubspace(sdkCtx, request.SubspaceId) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "subspace with id %d not found", request.SubspaceId)
+	}
+
+	sdkAddr, err := sdk.AccAddressFromBech32(request.User)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid user address: %s", request.User)
+	}
+
+	permission := k.GetUserPermissions(sdkCtx, request.SubspaceId, sdkAddr)
```

Comment:
The user might be in a group, should we show group permissions of the user?

Reply:
You're right, I've updated the response type to also include those permissions as well as the details of each one